### PR TITLE
Refactored Perks, with Help, other refinements

### PR DESCRIPTION
**Pull request by fearthecowboy**
Trimmed a few dependencies:
- perks/console
- yargs
- linq-es2015

2.0.* of perks/* - in separate repositories
 - extension install from git repo works now.
 - no more console patching
 - eliminated shell usage in extension loading 
 - much better folder locking 

Added:
 - @olydis new help stuff
 - new "poor-man's markdown" colorizer instead of `marked-terminal` (no file path assassination)
 - better colorization of errors 
 -

### CODE REVIEW

File: src/autorest/app.ts [hunk 1]
```diff
@@ -11,94 +11,65 @@ if (process.argv.indexOf("--no-upgrade-check") != -1) {
...
 import { isFile } from "@microsoft.azure/async-io";
-import { cli, enhanceConsole } from "@microsoft.azure/console";
-import { Exception, LazyPromise } from "@microsoft.azure/polyfill";
-import { Enumerable as IEnumerable, From } from "linq-es2015";
+import { Exception, LazyPromise } from "@microsoft.azure/tasks";
 import { networkEnabled, rootFolder, extensionManager, availableVersions, corePackage, installedCores, tryRequire, resolvePathForLocalVersion, ensureAutorestHome, selectVersion, pkgVersion } from "./autorest-as-a-service"
 import { gt } from "semver";
+import chalk from "chalk"
+
+function color(text: string): string {
```
Comment (fearthecowboy) [inline src/autorest/app.ts]: Discount-store :tm: markdown processor.

File: Samples/2a-validation/shell/stderr.txt [hunk 2]
```diff
@@ -1,3 +1,14 @@
+
```
Comment (fearthecowboy) [inline Samples/2a-validation/shell/stderr.txt]: I tweaked the colorization of the messages to make them pop a bit.

Comment (fearthecowboy) [reply Samples/2a-validation/shell/stderr.txt]: (Note the clever highlighting of the filename/row/column and the red/yellow for error/warning)

File: src/autorest-core/app.ts [hunk 3]
```diff
@@ -7,10 +7,36 @@ require('./static-loader.js').load(`${__dirname}/static_modules.fs`)
...
+
 require('events').EventEmitter.defaultMaxListeners = 100;
 process.env['ELECTRON_RUN_AS_NODE'] = "1";
 delete process.env['ELECTRON_NO_ATTACH_CONSOLE'];
 
+process.on("exit", () => {
+  Shutdown()
+});
+
+const color: (text: string) => string = (<any>global).color ? (<any>global).color : p => p;
```
Comment (fearthecowboy) [inline src/autorest-core/app.ts]: picks up `color` from the bootstrapper, falls back to NOP

File: src/autorest-core/package.json [hunk 4]
```diff
@@ -43,14 +43,14 @@
...
+    "static-link": "^0.2.3",
     "vscode-jsonrpc": "^3.3.1"
   },
   "static-link": {
     "entrypoints": [],
     "dependencies": {
-      "@microsoft.azure/async-io": "~1.0.22",
-      "@microsoft.azure/extension": "~1.2.12",
+      "@microsoft.azure/async-io": "~2.0.7",
+      "@microsoft.azure/extension": "^2.0.4",
```
Comment (fearthecowboy) [inline src/autorest-core/package.json]: try `--use=fearthecowboy/autorest-interactive` 💃

**Comment by olydis** [comment]
# 🤖 AutoRest automatic publish job 🤖
~~~ Haskell
> will publish once PR gets merged
~~~

**Pull request merged**